### ISSUE TITLE
Opus CUA planner: visual plan generation via browser exploration

### ISSUE THREAD

**Issue by mercurialsolo** [mercurialsolo/mantis#40]
## Problem

The current planner (Gemma4-E4B) converts plain text plans to task JSON without ever seeing the target website. It produces generic instructions like "click the listing card" that CUA models fail on because they lack:
- Visual anchors (what elements actually look like)
- Error handling (404, cookies, off-site navigation)
- Negative examples (what NOT to click)
- Layout-specific hints (where elements are positioned)

Hand-tuning these prompts has hit diminishing returns — 0/31 viable leads across $27 of GPU despite 10 bug fixes.

## Proposed Solution

Use Claude Opus as a **CUA-based visual planner** that actually browses the target site (via xdotool/Xvfb, same as execution) to generate plans.

### Two-phase approach:

**Phase 1: Opus explores (expensive, once)**
- Opus CUA navigates the target site through xdotool/Xvfb (same env as execution)
- It visually discovers: page layout, element positions, error states, popups
- Produces a rich task suite with visual anchors, error handlers, negative examples
- Does the non-looped parts of the plan (search setup, login, form discovery)
- Records its trajectory as training data for future fine-tuning

**Phase 2: Gemma4/EvoCUA executes loops (cheap, many times)**
- Takes the Opus-generated plan with all visual hints
- Runs the repetitive extraction loops (50 listings × N pages)
- Agentic learning loop refines within the run
- 10-100x cheaper per iteration than Opus

### Key insight
Opus doesn't just "write" the plan from text — it **browses the site** to discover:
- What listing cards actually look like on THIS site
- Where social icons / ads / menus are positioned
- What error pages look like (404, Cloudflare)
- How to handle cookies/popups on THIS site
- What the detail page layout is (where phone numbers appear)

This works for ANY website, not just BoatTrader, because Opus visually learns the layout at runtime.

### Architecture
```
User plain text plan
  → Opus CUA (xdotool/Xvfb) — explores site, discovers layout
    → Generates: rich task suite JSON + visual hints + error handlers
    → Also: records trajectory for Gemma4 fine-tuning (distillation)
  → Gemma4/EvoCUA (xdotool/Xvfb) — executes loops with Opus-generated hints
```

### Task Suite Schema (target output from Opus)
```json
{
  "tasks": [{
    "intent": "...",
    "visual_anchors": {
      "target": "LARGE rectangle ~250px tall with boat photo + price",
      "avoid": ["SMALL 20-40px colored squares (social)", "header dropdown"]
    },
    "error_handlers": [
      {"condition": "404 Page Not Found", "action": "Alt+Left, terminate SKIPPED"},
      {"condition": "Cookie popup", "action": "click Accept ONCE, max 2 steps"}
    ],
    "success_criteria": "Phone number (###-###-####) in Description section",
    "fast_fail_signals": ["404", "Page Not Found", "can't be reached"],
    "max_steps": 40
  }]
}
```

### GymRunner integration
- `error_handlers` and `fast_fail_signals` used by runner at runtime to nudge the model
- These are plan-generated, not hardcoded — works for any site
- Runner checks model's thinking against `fast_fail_signals` and injects early termination nudge

### Distillation bonus
Opus exploration trajectories (screenshot → action pairs) become training data for fine-tuning Gemma4-CUA on specific sites. Over time, Gemma4 learns site-specific patterns and needs less planning assistance.

## Files to create/modify
- `src/mantis_agent/gym/opus_planner.py` — new: Opus CUA exploration + plan generation
- `src/mantis_agent/gym/runner.py` — use `error_handlers` / `fast_fail_signals` from plan
- `src/mantis_agent/gym/workflow_runner.py` — pass plan metadata to runner
- `modal_cua_server.py` — new `--plan-with-opus` mode
- `training/convert_opus_trajectories.py` — new: convert Opus exploration to training data

## Cost estimate
- Opus exploration: ~$2-5 per site (one-time)
- Gemma4 execution: ~$0.50-1.00 per page (repeated)
- vs current: ~$5-10 per page with 0% hit rate

## Labels
enhancement, architecture, planning
